### DISCUSSION TITLE
Trino connect hive auth kerberos

### DISCUSSION THREAD

**Discussion by stbogdan77** [trinodb/trino#21226] [Q&A]
I have a problem connecting the trino to the hive
![image](https://github.com/prestodb/presto/assets/50769250/5845645f-6707-4c08-922e-955bfb91d8b0)

- connector.name=hive
- hive.metastore.uri=thrift://server5:9083
- hive.config.resources=/trino/etc/core-site.xml, /trino/etc/hdfs-site.xml, /trino/etc/hive-site.xml
- hive.metastore.authentication.type=KERBEROS
- hive.metastore.username=trino
- hive.metastore.thrift.impersonation.enabled=true
- hive.metastore.service.principal=krb55/GO.KDC@GO.KDC
- hive.metastore.client.principal=trino@GO.KDC
- hive.metastore.client.keytab=/trino/etc/key/trino.keytab
- hive.hdfs.authentication.type=KERBEROS
- hive.hdfs.impersonation.enabled=true
- hive.hdfs.trino.principal=trino@GO.KDC
- hive.hdfs.trino.keytab=/trino/etc/key/trino.keytab

If I try to connect via JDBC:POSTGRESQL, the config is like this and I don’t understand the error, if I open the database with these credentials everything works.

- connector.name=iceberg
- iceberg.compression-codec=gzip
- iceberg.catalog.type=jdbc
- iceberg.jdbc-catalog.catalog-name=test
- iceberg.jdbc-catalog.driver-class=org.postgresql.Driver
- iceberg.jdbc-catalog.connection-url=jdbc:postgresql://postgres1:5432/metastore
- iceberg.jdbc-catalog.connection-user=hive
- iceberg.jdbc-catalog.connection-password=**************
- iceberg.jdbc-catalog.default-warehouse-dir=hdfs://********//user/hive/warehouse

![image](https://github.com/prestodb/presto/assets/50769250/77aa4102-c8ff-489b-992f-28c41b245a64)
I'm just trying to see what tables are inside my directory

**Comment by Akanksha-kedia** [comment]
Kerberos uses the hive.metastore.client.principal and hive.metastore.client.keytab properties for authentication, so the hive.metastore.username property is unnecessary and can cause issues.

Here's how your configuration should look:

connector.name=hive
hive.metastore.uri=thrift://server5:9083
hive.config.resources=/trino/etc/core-site.xml, /trino/etc/hdfs-site.xml, /trino/etc/hive-site.xml
hive.metastore.authentication.type=KERBEROS
hive.metastore.thrift.impersonation.enabled=true
hive.metastore.service.principal=krb55/GO.KDC@GO.KDC
hive.metastore.client.principal=trino@GO.KDC
hive.metastore.client.keytab=/trino/etc/key/trino.keytab
hive.hdfs.authentication.type=KERBEROS
hive.hdfs.impersonation.enabled=true
hive.hdfs.trino.principal=trino@GO.KDC
hive.hdfs.trino.keytab=/trino/etc/key/trino.keytab.

**Comment by Akanksha-kedia** [comment]
hive.metastore.service.principal=hive/_HOST@YOUR-REALM.COM 
Note: The principal krb55/GO.KDC@GO.KDC you provided doesn't look like a typical Hive metastore service principal. Please ensure you're using the correct principal.

**Reply by akenO8** [reply]
@Akanksha-kedia May I ask, If front of the HMS have a LB proxy, what should the value of `hive.metastore.service.principal` should be?

**Comment by stbogdan77** [comment]
This is an error on the hivemetastore server in the logs
![Снимок экрана 2023-12-18 в 08 24 33](https://github.com/trinodb/trino/assets/50769250/33c4d3a0-1c23-4126-b755-c0b35b5bcd06)
